### PR TITLE
Backported Kafka scripts and image description

### PR DESCRIPTION
**Pull request by kyguy**

### CODE REVIEW

File: kafka/kafka-2.1.1/image.yaml [hunk 1]
```diff
@@ -1,40 +1,69 @@
...
+  - name: "COM_REDHAT_COMPONENT"
+    value: "amqstreams12-kafkabase-21-openshift-container"
+
 modules:
   repositories:
+    - name: cct_module
+      git:
+        url: https://github.com/jboss-openshift/cct_module.git
+        ref: sprint-28
     - path: ../modules
```
Comment (ppatierno) [inline kafka/kafka-2.1.1/image.yaml]: what do you think about giving it a name, I have used `modules` for the operator image for example.

File: kafka/kafka-2.1.1/image.yaml [hunk 2]
```diff
@@ -1,40 +1,69 @@
...
   install:
+    - name: jboss.container.user
+    - name: jboss.container.openjdk.jdk
+      version: "8"
+    - name: dynamic-resources
```
Comment (ppatierno) [inline kafka/kafka-2.1.1/image.yaml]: do we want to use the upstream `dynamic-resources.sh` script provided upstream? so removing this one as for the operator image?

File: kafka/kafka-2.2.0/help.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
-% amq7/amq-streams-kafka (1) Container Image Pages
+% amq7/amq-streams-kafka-2.1 (1) Container Image Pages
```
Comment (ppatierno) [inline kafka/kafka-2.2.0/help.md]: 2.2?

File: kafka/kafka-2.2.0/help.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
 
 # NAME
 
-Kafka - AMQ Streams image for running an Apache Kafka broker
+Kafka - AMQ Streams image for running Apache Kafka, Zookeeper, Kafka Connect and Mirror Maker - 2.1.1
```
Comment (ppatierno) [inline kafka/kafka-2.2.0/help.md]: 2.2.0?

File: kafka/kafka-2.2.0/image.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  - name: "COM_REDHAT_COMPONENT"
+    value: "amqstreams12-kafkabase-22-openshift-container"
+
+modules:
+  repositories:
+    - name: cct_module
+      git:
+        url: https://github.com/jboss-openshift/cct_module.git
+        ref: sprint-28
+    - path: ../modules
```
Comment (ppatierno) [inline kafka/kafka-2.2.0/image.yaml]: same as above, using `modules` as name as well?

File: kafka/modules/kafka/install.sh [hunk 6]
```diff
@@ -2,9 +2,39 @@
...
 SCRIPTS_DIR=${SCRIPT_DIR}/scripts
+STUNNEL_DIR=${SCRIPT_DIR}/stunnel-scripts
+S2I_DIR=${SCRIPT_DIR}/s2i-scripts
+LICENSE_DIR=/root/licenses
+PRODUCT_LICENSE_DIR=${LICENSE_DIR}/${COM_REDHAT_COMPONENT}
+
+mkdir $KAFKA_HOME
+mkdir $STUNNEL_HOME
+mkdir $S2I_HOME
+mkdir -p -m g+rw /usr/local/var/run/
```
Comment (ppatierno) [inline kafka/modules/kafka/install.sh]: from where this line comes out?

File: kafka/modules/kafka/install.sh [hunk 7]
```diff
@@ -2,9 +2,39 @@
...
+
+tar xvfz "${SOURCES_DIR}/kafka.tar.gz" -C ${KAFKA_HOME} --strip-components=1
+
+mkdir /opt/prometheus
+mkdir /opt/prometheus/config
+mv ${SOURCES_DIR}/jmx_prometheus_javaagent.jar /opt/prometheus/jmx_prometheus_javaagent.jar
+mv ${SOURCES_DIR}/kafka-agent-* ${KAFKA_HOME}/libs
```
Comment (ppatierno) [inline kafka/modules/kafka/install.sh]: didn't we agree on using `cp -r` instead of `mv`?

File: kafka/modules/kafka/scripts/dynamic_resources.sh [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (ppatierno) [inline kafka/modules/kafka/scripts/dynamic_resources.sh]: Ok so you imported this file from upstream, using for Kafka 2.2 but not for the 2.1 as mentioned in one of the previous comments.

File: kafka/modules/kafka/install.sh (hunk 6 above)
Comment (kyguy) [reply kafka/modules/kafka/install.sh]: Its a read/writable directory for stunnel sockets: https://github.com/strimzi/strimzi-kafka-operator/blob/master/docker-images/kafka/Dockerfile#L52

File: kafka/kafka-2.1.1/image.yaml [hunk 9]
```diff
@@ -1,40 +1,69 @@
...
-from: amq7/amq-streams-kafka-base:1.1.0-kafka-2.1.1
+name: amq7/amq-streams-kafka-2.1
+description: "AMQ Streams image for running Apache Kafka, Zookeeper, Kafka Connect and Mirror Maker"
+version: "1.2.0"
+from: rhel7:7-released
 
 labels:
   - name: "com.redhat.component"
-    value: "amqstreams11-kafka-211-openshift-container"
+    value: "amqstreams12-kafka-21-openshift-container"
```
[the remark this answers is not in the document]
Comment (kyguy) [reply kafka/kafka-2.1.1/image.yaml]: As we chatted early, we won't track it explicitly in these names so we can gain more flexibility with our patches ( Being able to use the same branches and brew packages for micros releases of Kakfa )

File: kafka/modules/kafka/scripts/dynamic_resources.sh (hunk 8 above)
Comment (kyguy) [reply kafka/modules/kafka/scripts/dynamic_resources.sh]: fixed

**Comment by kyguy** [comment]
Alright, pushed some changes and addressed some of the comments. Let me know what you think

**Pull request merged**